code/modules/chemistry/Chemistry-Machinery.dm

```diff
@@ -483,6 +483,26 @@
 			logTheThing("combat",usr,null,"used the [src.name] to create [bottlename] bottle containing [log_reagents(B)] at log_loc[src].")
 			return
 
+		else if (href_list["createcan"])
+			var/default = R.get_master_reagent_name()
+			var/input_name = input(usr, "Name the can:", "Name", default) as null|text
+			if(input_name && input_name != default)
+				phrase_log.log_phrase("bottle", input_name, no_duplicates=TRUE)
+			var/bottlename = copytext(html_encode(input_name), 1, 32)
+
+			var/input_design = input(usr, "Choose the design (1~26):", "Design", default) as null|num
+
+			if (isnull(bottlename) || !src.beaker || !R || !length(bottlename) || bottlename == " " || get_dist(usr, src) > 1 || isnull(input_design) || input_design > 26 || input_design < 1)
+				return
```

Comment:
```suggestion
			if (!src.beaker || !R || !length(bottlename) || bottlename == " " || get_dist(usr, src) > 1 || isnull(input_design) || input_design > 26 || input_design < 1)
				return
```
`length(null)` is 0

Reply:
happy to take it out if you want, it's identical with the bottling code i lazily copied over and it seems a little dodgy to rely on the length of nulls, that's just the C# talkin though